pkg/processor/procbuiltin/unwrap.go

```diff
@@ -93,6 +96,172 @@ func (p *unwrapProcessor) Process(_ context.Context, in record.Record) (record.R
 	return out, nil
 }
 
+/*
+Example of an OpenCDC record:
+{
+  "key": "NWQ0N2UwZGQtNTkxYi00MGEyLTk3YzMtYzc1MDY0MWU3NTc1",
+  "metadata": {
+    "conduit.source.connector.id": "source-generator-78lpnchx7tzpyqz:source",
+    "opencdc.readAt": "1706028881541916000",
+    "opencdc.version": "v1"
+  },
+  "operation": "create",
+  "payload": {
+    "after": {
+      "event_id": 2041181862,
+      "msg": "string 4c88f20f-aa77-4f4b-9354-e4fdb1989a52",
+      "pg_generator": false,
+      "sensor_id": 54434691,
+      "triggered": false
+    },
+    "before": null
+  },
+  "position": "ZWIwNmJiMmMtNWNhMS00YjUyLWE2ZmMtYzc0OTFlZDQ3OTYz"
+}
+*/
+
+// openCDCUnwrapper unwraps an OpenCDC record from the payload, by unmarhsalling rec.Payload.After into type Record.
+type openCDCUnwrapper struct{}
+
+// UnwrapOperation extracts operation from a structuredData record.
+func (o *openCDCUnwrapper) UnwrapOperation(structData record.StructuredData) (record.Operation, error) {
+	var operation record.Operation
+	op, ok := structData["operation"]
+	if !ok {
+		return operation, cerrors.Errorf("record payload after doesn't contain operation")
+	}
+
+	switch o := op.(type) {
+	case record.Operation:
+		operation = o
+	case string:
+		err := operation.UnmarshalText([]byte(o))
+		if err != nil {
+			return operation, cerrors.Errorf("couldn't unmarshal record operation")
+		}
+	}
+	return operation, nil
+}
+
+// UnwrapMetadata extracts metadata from a structuredData record.
+func (o *openCDCUnwrapper) UnwrapMetadata(structData record.StructuredData) (record.Metadata, error) {
+	var metadata record.Metadata
+	meta, ok := structData["metadata"]
+	if !ok {
+		return metadata, cerrors.Errorf("record payload after doesn't contain metadata")
+	}
+
+	switch m := meta.(type) {
+	case record.Metadata:
+		metadata = m
+	case map[string]interface{}:
+		metadata = make(record.Metadata, len(m))
+		for k, v := range m {
+			metadata[k] = fmt.Sprint(v)
+		}
+	}
+	return metadata, nil
+}
+
+// UnwrapKey extracts key from a structuredData record.
+func (o *openCDCUnwrapper) UnwrapKey(structData record.StructuredData) (record.Data, error) {
+	var key record.Data
+	ky, ok := structData["key"]
+	if !ok {
+		return key, cerrors.Errorf("record payload after doesn't contain key")
+	}
+
+	switch k := ky.(type) {
+	case record.Data:
+		key = k
+	case string:
```

Comment:
is it possible for a key to be an `int`?

Reply:
@samirketema My assumption is that data will be either formatted as `record.Data` or part of a JSON string. I'll add a case on default to return an error so we can catch those in case they occur.